roadmap/implementers-guide/src/node/backing/statement-distribution.md

```diff
@@ -1,107 +1,401 @@
+   - `RequestManager::dispatch_requests` sends any queued-up requests.
+      - Calls `RequestManager::next_request` to completion.
+        - Creates the `OutgoingRequest`, saves the receiver in
+          `RequestManager::pending_responses`.
+      - Does nothing if we have more responses pending than the limit of parallel
+        requests.
+
+2. Peer
+
+   - Requests come in on a peer on the `IncomingRequestReceiver`.
+     - Runs in a background responder task which feeds requests to `answer_request`
+       through `MuxedMessage`.
+     - This responder task has a limit on the number of parallel requests.
+   - `answer_request` on the peer takes the request and sends a response.
+     - Does this using the response sender on the request.
+
+3. Requesting Validator
+
+   - `receive_response` on the original validator yields a response.
+     - Response was sent on the request's response sender.
+     - Uses `RequestManager::await_incoming` to await on pending responses in an
+       unordered fashion.
+     - Runs on the `MuxedMessage` receiver.
+   - `handle_response` handles the response.
+
+### API
+
+- `dispatch_requests`
+  - Dispatches pending requests for candidate data & statements.
+- `answer_request`
+  - Answers an incoming request for a candidate.
+  - Takes an incoming `AttestedCandidateRequest`.
+- `receive_response`
+  - Wait on the next incoming response.
+  - If there are no requests pending, this future never resolves.
+  - Returns `UnhandledResponse`
+- `handle_response`
+  - Handles an incoming response.
+  - Takes `UnhandledResponse`
+
+## Manifests
+
+A manifest is a message about a known backed candidate, along with a description
+of the statements backing it. It can be one of two kinds:
+
+- `Full`: Contains information about the candidate and should be sent to peers
+  who may not have the candidate yet.
+- `Acknowledgement`: Omits information implicit in the candidate, and should be
+  sent to peers which are guaranteed to have the candidate already.
+
+### Manifest Exchange
+
+Manifest exchange is when a receiving node received a `Full` manifest and
+replied with an `Acknowledgement`. It indicates that both nodes know the
+candidate as valid and backed. This allows the nodes to send `Statement`
+messages directly to each other for any new statements.
+
+Why? This limits the amount of statements we'd have to deal with w.r.t.
+candidates that don't really exist. Limiting out-of-group statement distribution
+between peers to only candidates that both peers agree are backed and exist,
+ensures we only have to store statements about real candidates.
+
+In practice, manifest exchange means that one of three things have happened:
+
+- They announced, we acknowledged
+- We announced, they acknowledged
+- We announced, they announced (not sure if this can actually happen; it would
+  happen if 2 nodes had each other in their sending set and they sent manifests
+  at the same time. The code accounts for this anyway)
```

Comment:
This should be stated clearly before the merge, looks like a draft leftover

Reply:
@rphmeier Can you please help me clarify this line?

Reply:
It should not be possible for 2 nodes to have each other in their sending set, so the line can be clarified on that basis.